ee/tabby-webserver/src/hub/mod.rs

```diff
@@ -50,6 +51,29 @@ pub(crate) async fn ws_handler(
         .into_response()
 }
 
+pub(crate) async fn require_login_middleware(
```

Comment:
this shouldn't be part of `hub` crate.

I suggest move the function to `src/handler.rs` and orchestrate the middleware there.